extension/basicauthextension/extension.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basicauthextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/basicauthextension"
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/tg123/go-htpasswd"
+	"go.opentelemetry.io/collector/client"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
+)
+
+type basicAuth struct {
+	htpasswd  HtpasswdSettings
+	matchFunc func(username, password string) bool
+}
+
+func newExtension(cfg *Config) configauth.ServerAuthenticator {
+	ba := basicAuth{
+		htpasswd: cfg.Htpasswd,
+	}
+	return configauth.NewServerAuthenticator(configauth.WithStart(ba.start), configauth.WithAuthenticate(ba.authenticate))
+}
+
+func (ba *basicAuth) start(ctx context.Context, host component.Host) error {
+	var rs []io.Reader
+
+	if ba.htpasswd.File != "" {
```

Comment:
What happens when no inline data nor file is provided? Will it just reject all requests? I think it should be considered a config error.

Reply:
It will reject all requests. Can make it fail as you suggested.

Reply:
I think it will be more appropriate.

Reply:
Added the check during the creation of the extension.